ibm/resource_ibm_is_subnet_routing_table_attachment.go

```diff
@@ -0,0 +1,316 @@
+			},
+			isRoutingTableResourceType: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The resource type",
+			},
+
+			rtRouteTransitGatewayIngress: {
+				Type:        schema.TypeBool,
+				Computed:    true,
+				Description: "If true, this routing table will be used to route traffic that originates from Transit Gateway to this VPC.",
+			},
+
+			rtRouteVPCZoneIngress: {
+				Type:        schema.TypeBool,
+				Computed:    true,
+				Description: "If true, this routing table will be used to route traffic that originates from subnets in other zones in this VPC.",
+			},
+
+			rtSubnets: {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						rtName: {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "Subnet name",
+						},
+						"id": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "Subnet ID",
+						},
+					},
+				},
+			},
+
+			rtRoutes: {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						rtName: {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "route name",
+						},
+						"id": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "route ID",
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourceIBMISSubnetRoutingTableAttachmentCreate(d *schema.ResourceData, meta interface{}) error {
+	sess, err := vpcClient(meta)
+	if err != nil {
+		return err
+	}
+
+	subnet := d.Get(isSubnetID).(string)
+	routingTable := d.Get(isRoutingTableID).(string)
+
+	// Construct an instance of the RoutingTableIdentityByID model
+	routingTableIdentityModel := new(vpcv1.RoutingTableIdentityByID)
+	routingTableIdentityModel.ID = &routingTable
+
+	// Construct an instance of the ReplaceSubnetRoutingTableOptions model
+	replaceSubnetRoutingTableOptionsModel := new(vpcv1.ReplaceSubnetRoutingTableOptions)
+	replaceSubnetRoutingTableOptionsModel.ID = &subnet
+	replaceSubnetRoutingTableOptionsModel.RoutingTableIdentity = routingTableIdentityModel
+	resultACL, response, err := sess.ReplaceSubnetRoutingTable(replaceSubnetRoutingTableOptionsModel)
+
+	if err != nil {
+		log.Printf("[DEBUG] Error while attaching a routing table to a subnet %s\n%s", err, response)
+		return fmt.Errorf("Error while attaching a routing table to a subnet %s\n%s", err, response)
+	}
+	d.SetId(subnet)
+	log.Printf("[INFO] Routing Table : %s", *resultACL.ID)
+	log.Printf("[INFO] Subnet ID : %s", subnet)
+
+	return resourceIBMISSubnetRoutingTableAttachmentRead(d, meta)
+}
+
+func resourceIBMISSubnetRoutingTableAttachmentRead(d *schema.ResourceData, meta interface{}) error {
+	id := d.Id()
+	sess, err := vpcClient(meta)
+	if err != nil {
+		return err
+	}
+
+	getSubnetRoutingTableOptionsModel := &vpcv1.GetSubnetRoutingTableOptions{
```

Comment:
if we make a composite ID, we can split the id and use subnet ID here to support the terraform import